modules/ui.py

```diff
@@ -1,29 +1,23 @@
-import html
 import json
-import math
 import mimetypes
 import os
-import platform
-import random
 import sys
-import tempfile
-import time
 import traceback
-from functools import partial, reduce
+from functools import reduce
 import warnings
 
 import gradio as gr
 import gradio.routes
 import gradio.utils
 import numpy as np
-from PIL import Image, PngImagePlugin
+from PIL import Image, PngImagePlugin  # noqa: F401

```

Comment:
Curious as to why this needs an unused-import noqa, but I can look at that later.

Reply:
There was issue with another part of PIL breaking things unless this plugin is specifically imported before it. I don't fully recall it and this may not be what triggered it, but I kept it in just to be safe.